backend/handlers/math/outcomes/dbpm/marketshares_test.go

```diff
@@ -1,551 +1,709 @@
+				util.GenerateBet(10, "YES", "two", 1, time.Minute),
+			},
+			ProbabilityChanges: generateProbability(0.500, 0.167, 0.375),
+			ExpectedPayouts: generateCoursePayouts(
+				[]float64{4.1600000000000001, 0},
+				[]string{"NO", "YES"},
+			),
+		},
+		{
+			Name: "ThirdBetYesDirection",
+			Bets: []models.Bet{
+				util.GenerateBet(20, "NO", "one", 1, 0),
+				util.GenerateBet(10, "YES", "two", 1, time.Minute),
+				util.GenerateBet(10, "YES", "three", 1, 2*time.Minute),
+			},
+			ProbabilityChanges: generateProbability(0.500, 0.167, 0.375, 0.500),
+			ExpectedPayouts: generateCoursePayouts(
+				[]float64{6.6599999999999993, 1.25, 0},
+				[]string{"NO", "YES", "YES"},
+			),
+		},
+		{
+			Name: "FourthBetNegativeNoDirection",
+			Bets: []models.Bet{
+				util.GenerateBet(20, "NO", "one", 1, 0),
+				util.GenerateBet(10, "YES", "two", 1, time.Minute),
+				util.GenerateBet(10, "YES", "three", 1, 2*time.Minute),
+				util.GenerateBet(-10, "NO", "one", 1, 3*time.Minute),
 			},
+			ProbabilityChanges: generateProbability(0.500, 0.167, 0.375, 0.500, 0.625),
+			ExpectedPayouts: generateCoursePayouts(
+				[]float64{9.1600000000000001, 2.5, 1.25, 0},
+				[]string{"NO", "YES", "YES", "NO"},
+			),
 		},
 	}
+
 	for _, tc := range testcases {
 		t.Run(tc.Name, func(t *testing.T) {
-			result := AggregateUserPayoutsDBPM(tc.Bets, tc.AdjustedScaledPayouts)
+			actualPayouts := CalculateCoursePayoutsDBPM(tc.Bets, tc.ProbabilityChanges)
 
-			// Create a map for expected results for easy lookup
-			expectedResults := make(map[string]MarketPosition)
-			for _, pos := range tc.AggregatedPositions {
-				expectedResults[pos.Username] = pos
+			// Log all actual payouts for debugging
+			for i, payout := range actualPayouts {
+				t.Logf(
+					"%s: Actual payout %d -> {Payout: %.17g, Outcome: %s}",
+					tc.Name, i, payout.Payout, payout.Outcome,
+				)
 			}
 
-			// Create a map from the results for comparison
-			resultsMap := make(map[string]MarketPosition)
-			for _, pos := range result {
-				resultsMap[pos.Username] = pos
+			if len(actualPayouts) != len(tc.ExpectedPayouts) {
+				t.Fatalf("%s: expected %d payouts, got %d", tc.Name, len(tc.ExpectedPayouts), len(actualPayouts))
 			}
 
-			// Check if the results match expected results for each user
-			for username, expectedPos := range expectedResults {
-				resultPos, ok := resultsMap[username]
-				if !ok {
-					t.Errorf("Test %s failed: missing position for username %s", tc.Name, username)
-					continue
-				}
-				if resultPos.YesSharesOwned != expectedPos.YesSharesOwned || resultPos.NoSharesOwned != expectedPos.NoSharesOwned {
-					t.Errorf("Test %s failed for %s: expected %+v, got %+v", tc.Name, username, expectedPos, resultPos)
+			for i, payout := range actualPayouts {
+				expected := tc.ExpectedPayouts[i]
+
+				// Log both expected and actual for each mismatch
+				if payout.Payout != expected.Payout || payout.Outcome != expected.Outcome {
+					t.Errorf(
```

Comment:
This will always print when running tests, it may be cleaner to just have the debug output in the error statement now that this test is written.